internal/lifecycle/lifecycle_test.go

```diff
@@ -220,6 +225,40 @@ func TestLifecycleStop(t *testing.T) {
 	})
 }
 
+func TestHookRecordsFormat(t *testing.T) {
+	t.Run("SortRecords", func(t *testing.T) {
+		r := make(HookRecords, 2)
+		t1, _ := time.ParseDuration("10ms")
+		t2, _ := time.ParseDuration("20ms")
```

Comment:
we should handle these errors.
```suggestion
		t1, err := time.ParseDuration("10ms")
		require.NoError(t, err)
		t2, err := time.ParseDuration("20ms")
		require.NoError(t, err)
```